src/FSharpLint.Core/Rules/Conventions/NoPartialFunctions.fs

```diff
@@ -119,94 +119,104 @@ let rec private tryFindTypedExpression (range: Range) (expression: FSharpExpr) =
+            | FSharpExprPatterns.Sequential(firstExpr, secondExpr) -> 
+                tryFindTypedExpression range firstExpr |> Option.orElse (tryFindTypedExpression range secondExpr)
+            | FSharpExprPatterns.TryFinally(bodyExpr, finalizeExpr, _, _) -> 
+                tryFindTypedExpression range bodyExpr |> Option.orElse (tryFindTypedExpression range finalizeExpr)
+            | FSharpExprPatterns.TryWith(bodyExpr, _, _, _catchVar, catchExpr, _, _) -> 
+                tryFindTypedExpression range bodyExpr |> Option.orElse (tryFindTypedExpression range catchExpr)
+            | FSharpExprPatterns.TupleGet(_tupleType, _tupleElemIndex, tupleExpr) -> 
+                tryFindTypedExpression range tupleExpr
+            | FSharpExprPatterns.DecisionTree(decisionExpr, decisionTargets) -> 
+                tryFindTypedExpression range decisionExpr
+                |> Option.orElse (decisionTargets |> Seq.choose (fun (_, expr) -> tryFindTypedExpression range expr) |> Seq.tryHead)
+            | FSharpExprPatterns.DecisionTreeSuccess (_decisionTargetIdx, decisionTargetExprs) -> 
+                tryFindFirst decisionTargetExprs
+            | FSharpExprPatterns.TypeLambda(_genericParam, bodyExpr) -> 
+                tryFindTypedExpression range bodyExpr
+            | FSharpExprPatterns.TypeTest(_ty, inpExpr) -> 
+                tryFindTypedExpression range inpExpr
+            | FSharpExprPatterns.UnionCaseSet(unionExpr, unionType, unionCase, unionCaseField, valueExpr) -> 
+                tryFindTypedExpression range unionExpr |> Option.orElse (tryFindTypedExpression range valueExpr)
+            | FSharpExprPatterns.UnionCaseGet(unionExpr, _unionType, _unionCase, _unionCaseField) -> 
+                tryFindTypedExpression range unionExpr
+            | FSharpExprPatterns.UnionCaseTest(unionExpr, _unionType, _unionCase) -> 
+                tryFindTypedExpression range unionExpr
+            | FSharpExprPatterns.UnionCaseTag(unionExpr, _unionType) -> 
+                tryFindTypedExpression range unionExpr
+            | FSharpExprPatterns.ObjectExpr(_objType, baseCallExpr, overrides, interfaceImplementations) -> 
+                let interfaceImlps = interfaceImplementations |> List.collect snd
+                baseCallExpr :: (List.append overrides interfaceImlps |> Seq.cast<FSharpExpr> |> Seq.toList)
+                |> tryFindFirst
+            | FSharpExprPatterns.TraitCall(_sourceTypes, _traitName, _typeArgs, _typeInstantiation, _argTypes, argExprs) -> 
+                tryFindFirst argExprs
+            | FSharpExprPatterns.ValueSet(_valToSet, valueExpr) -> 
+                tryFindTypedExpression range valueExpr
+            | FSharpExprPatterns.WhileLoop(guardExpr, bodyExpr, _) -> 
+                tryFindTypedExpression range guardExpr |> Option.orElse (tryFindTypedExpression range bodyExpr)
+            | _ -> None
+        with
+        | _ ->
```

Comment:
@webwarrior-ws a generic catch is dangerous so let's check the ex.Message, even if ugly

Reply:
done